zilliqa/src/consensus.rs

```diff
@@ -325,28 +341,126 @@ impl Consensus {
         Ok(None)
     }
 
-    fn download_blocks_up_to(&mut self, to: u64) -> Result<()> {
-        for view in (self.view + 1)..to {
-            self.view = view;
-            self.block_store.request_block_by_view(view)?;
-        }
-        self.view = to + 1;
+    pub fn head_block(&self) -> Block {
```

Comment:
Head block is now prioritized over finalized block for use. 

I think asking the DB here for the highest block number will become cleaner with james' DB pr